### PR TITLE
UI fixes on the Onboarding Screens and the bottom navBar

### PR DESCRIPTION
**Pull request by kamdaou-amalitech**
Used constraint layout to create onboarding screens
Added code to avoid having the dots moving while sliding
Moved the badge that is in the bottom nav bar on top of the notifications icon

### CODE REVIEW

**Review by bensek — APPROVED** [review]
Great

File: ui/onboarding/src/main/java/com/amalitech/onboarding/OnboardingScreen.kt [hunk 1]
```diff
@@ -183,36 +190,46 @@ fun OnBoard(
...
             }
-            Spacer(modifier = Modifier.height(spacing.spaceSmall))
             ImageWithLegend(
                 title = if (title != null) stringResource(id = title) else null,
                 description = stringResource(id = description),
                 painter = painter,
             )
+        }
+
+        if (selectedIndex != 3) {
```
Comment (esthcarelle) [inline ui/onboarding/src/main/java/com/amalitech/onboarding/OnboardingScreen.kt:205]: We should not have the 3 hard coded please, try something like this if  selectedIndex != arraySize.length - 1 show the sliding Dot else show the default button

File: ui/onboarding/src/main/java/com/amalitech/onboarding/components/SlidingDots.kt [hunk 2]
```diff
@@ -17,20 +16,20 @@ import androidx.compose.ui.tooling.preview.Preview
...
     modifier: Modifier = Modifier,
     selectedIndex: Int = 1
 ) {
-    Row(
-        modifier = modifier.fillMaxWidth(),
-        horizontalArrangement = Arrangement.SpaceBetween,
-    ) {
+    val spacing = LocalSpacing.current
+    Row(modifier = modifier) {
         (0..3).forEach {
```
Comment (esthcarelle) [inline ui/onboarding/src/main/java/com/amalitech/onboarding/components/SlidingDots.kt:28]: Can we avoid having this 3 hard coded, let s pass it as argument

**Review by esthcarelle — CHANGES_REQUESTED** [review]

File: ui/onboarding/src/main/java/com/amalitech/onboarding/components/SlidingDots.kt (hunk 2 above)
Comment (kamdaou-amalitech) [reply ui/onboarding/src/main/java/com/amalitech/onboarding/components/SlidingDots.kt:28]: Ok

**Review by esthcarelle — APPROVED** [review]

**Comment by esthcarelle** [comment]
@kamdaou-amalitech could you please remove this  OnboardingScreen {

                } in MainActivity before I merge it

**Comment by kamdaou-amalitech** [comment]
@esthcarelle it's done
I have also put the badge on the icon in the bottom nav bar as requested by Evans
Screenshot_20230612_125916_com amalitech bookmeetingroom production debug

**Pull request merged**